### PR TITLE
Fixes Issue#5853 Tapping a "... sent you an email" notification has no effect

### LINKED ISSUE #5853: Tapping a "... sent you an email" notification has no effect
Should open web browser on relevant URL.
Currently it just marks as read.

https://github.com/user-attachments/assets/50a641e7-2315-4e14-ac09-4665d31dd55d

### PR DESCRIPTION
**Pull request by ErenratZeng**
This change improves the user experience by ensuring email notifications are properly categorized and provides useful feedback when they are clicked.

**Description (required)**

Fixes #5853

What changes did you make and why?

This Pull request adds logic to classify notifications as "email" type when the notification text contains "sent you an email". This change ensures that email notifications are properly categorized, which was previously missing, leading to all email notifications being marked as UNKNOWN type.

Previously, email-related notifications from the backend were missing a URL. As a result, when users clicked on these notifications, there was no link to open, and the notifications were categorized as UNKNOWN. This led to a poor user experience since there was no feedback provided when the user clicked on an email notification.

Changes:
- **NotificationClient**:
  - Modified `WikimediaNotification.toCommonsNotification()` to check if the notification text contains "sent you an email". If it does, the notification is classified as `EMAIL_MESSAGE` instead of the default `UNKNOWN`.

- **NotificationActivity**:
  - In the `NotificatinAdapter` click handler, added a check for `EMAIL_MESSAGE` type. When an email-type notification is clicked, a "Check your mail box" prompt is shown using `Snackbar`, instead of attempting to open a URL (which is typically missing for such notifications).

- **NotificationType**:
  - Added a new `EMAIL` type to categorize email-related notifications.

**Tests performed (required)**

Tested {build variant} on {Galaxy s20 SM-G9810} with API level {33}.

**Screenshots (for UI changes only)**
Screenshot_20241017_100941_Commons

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java [hunk 1]
```diff
@@ -147,8 +148,11 @@ private void initListView() {
             refresh(false);
         }
         adapter = new NotificatinAdapter(item -> {
-            Timber.d("Notification clicked %s", item.getLink());
-            handleUrl(item.getLink());
+            if (item.getNotificationType() == NotificationType.EMAIL){
+                ViewUtil.showLongSnackbar(binding.container, "Check your mail box");
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java:152]: Better write "email inbox" fully to make it 100% explicit.
Also, can you internationalize the string?

File: app/src/main/java/fr/free/nrw/commons/notification/NotificationClient.kt [hunk 2]
```diff
@@ -21,43 +22,55 @@ class NotificationClient
...
+            } else {
+                Observable.just(false)
             }
+        }
+
+    private fun WikimediaNotification.toCommonsNotification(): Notification {
+
+        val notificationText = contents?.compactHeader ?: ""
+
+        val notificationType = if (notificationText.contains("sent you an email", ignoreCase = true)){
```
[the remark this answers is not in the document]
Comment (ErenratZeng) [reply app/src/main/java/fr/free/nrw/commons/notification/NotificationClient.kt:59]: Yep, It works on both Chinese(Simplified) and Japanese. Because the notification page was not being translated.

File: app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java (hunk 1 above)
Comment (ErenratZeng) [reply app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java:152]: All good, but Can I have some clarifications for "internationalize the string".

Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java:152]: The snackbar should be shown in French to users who use the app in French, for instance. :-)
https://developer.android.com/guide/topics/resources/localization

Example: https://github.com/commons-app/apps-android-commons/blob/main/app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java#L104

**Comment by ErenratZeng** [comment]
Now It should work as internationalize string

File: app/src/main/java/fr/free/nrw/commons/notification/NotificationClient.kt [hunk 3]
```diff
@@ -21,43 +22,55 @@ class NotificationClient
         @param:Named(NetworkingModule.NAMED_COMMONS_CSRF) private val csrfTokenClient: CsrfTokenClient,
         private val service: NotificationInterface,
     ) {
-        fun getNotifications(archived: Boolean): Single<List<Notification>> =
+    fun getNotifications(archived: Boolean): Single<List<Notification>> =
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/notification/NotificationClient.kt:25]: What is the purpose of this change of a large number of lines? If it is indentation, better do it in a separate pull request, to make the changes in this one easier to spot. Thanks!

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Otherwise looks good, and works perfectly. 🙂

File: app/src/main/java/fr/free/nrw/commons/notification/NotificationClient.kt (hunk 3 above)
Comment (ErenratZeng) [reply app/src/main/java/fr/free/nrw/commons/notification/NotificationClient.kt:25]: I think there might be some Android studio version format issues for the indentation, there was why it changed. There was weird, I did not notice that when I pushing.

Comment (ErenratZeng) [reply app/src/main/java/fr/free/nrw/commons/notification/NotificationClient.kt:25]: Do I still need to change it back to the old one? I afraid of making more mess on it.

Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/notification/NotificationClient.kt:25]: If you are afraid of messing things up, the safest might be to create a new branch from main, and apply your changes in that new branch, then send a new pull request based on it.

Thanks a lot! 🙂

Comment (ErenratZeng) [reply app/src/main/java/fr/free/nrw/commons/notification/NotificationClient.kt:25]: Got that, I gonna create a new pull request for that.
Cheers
